### ISSUE TITLE
A few feature requests for the new Twitch name changing feature

### ISSUE THREAD

**Issue by Jacky1356400** [chatty/chatty#128]
Twitch added a new feature, as quite a lot of people knows about, [name changing](https://blog.twitch.tv/the-hype-is-real-name-changes-are-here-c629c4697154?sf58352669=1).

This feature is good, but also ruins chat logging and in searching messages from the same user.
From what I know so far, name change keeps `user-id` in IRCv3 tags the same.

It would be great if there's the following enhancements:
-User Dialog includes the User ID and keeps the messages from the same person before and after the name change during the session Chatty is running
-Includes option to log the User ID as part of the log to file feature
-Shows the previous name of a user after they changed the name (Not fully possible yet but thought I'd include this in case Twitch implements this into part of API)

**Comment by tduva** [comment]
Logging the UserID is probably the easiest of them all. Keeping the message history in the User Dialog after a name change would be a bit harder. I can't completely rely on UserIDs since not everything in chat has them. I'll have to think about this (and wait what Twitch does).

**Comment by Jacky1356400** [comment]
Yeah... Twitch hasn't really implement anything about the name change system into the API apart from User IDs.
Hopefully Twitch will sort out some solutions for that.

**Comment by Jacky1356400** [comment]
Update: Apparently there's now [User IDs for Mod Action PubSub](https://discuss.dev.twitch.tv/t/username-changes-identifying-users/9373/35), so it would make logging User IDs for Mod Actions a lot easier.

**Comment by tduva** [comment]
Since the latest beta you can now customize regular chat messages for logging chat to file to include the user id.
